### PR TITLE
[dataprotection] Add support for Cross Region Restore for Backup Vaults

### PR DESCRIPTION
**Pull request by zubairabid**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

* Added support for Cross Region Restore for Backup Vaults.
* `az dataprotection backup-vault create`: New parameter `--cross-region-restore-state/--crr-state` that can be set to Enabled/Disabled.
* `az dataprotection backup-vault update`: New parameter `--cross-region-restore-state/--crr-state` that can be set to Enabled/Disabled.
* `az dataprotection backup-vault list-from-resourcegraph`: New command to fetch Backup Vault details from Azure Resource Graph.
* `az dataprotection recovery-point list`: New parameter `--use-secondary-region` to be used when listing from the secondary region.
* `az dataprotection backup-instance validate-for-restore`: New parameter `--use-secondary-region` to be used when restoring to the secondary region.
* `az dataprotection backup-instance restore trigger`: New parameter `--use-secondary-region` to be used when restoring to the secondary region.
* `az dataprotection backup-job list`: New parameter `--use-secondary-region` which can be used in disaster scenario when primary region is down.
* `az dataprotection backup-job show`: New parameter `--use-secondary-region` which can be used in disaster scenario when primary region is down.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
dataprotection

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by zubairabid** [comment]
> Please fix CI issues

all CI issues have been fixed, could you please re-run the pipeline so the Cred scan passes this time?

**Comment by zubairabid** [comment]
@yonzhan @evelyn-ys can you please review and merge the PR?

File: src/dataprotection/setup.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 from setuptools import setup, find_packages
 
 # HISTORY.rst entry.
-VERSION = '0.11.2'
+VERSION = '0.12.0'
```
Comment (evelyn-ys) [inline src/dataprotection/setup.py:13]: Please check the version to follow https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md#versioning-practice

File: src/dataprotection/azext_dataprotection/manual/commands.py [hunk 2]
```diff
@@ -53,24 +59,34 @@ def load_command_table(self, _):
...
+        # TriggerCRR as CRRTriggerRestore,
     )
     self.command_table['dataprotection backup-instance create'] = BackupInstanceCreate(loader=self)
-    self.command_table['dataprotection backup-instance validate-for-restore'] = BackupInstanceValidateRestore(loader=self)
-    self.command_table['dataprotection backup-instance restore trigger'] = BackupInstanceRestoreTrigger(loader=self)
+    # self.command_table['dataprotection backup-instance validate-for-restore'] = BackupInstanceValidateRestore(loader=self)
+    # self.command_table['dataprotection backup-instance restore trigger'] = BackupInstanceRestoreTrigger(loader=self)
+
+    # self.command_table['dataprotection cross-region-restore validate'] = CRRValidateRestore(loader=self)
+    # self.command_table['dataprotection cross-region-restore trigger'] = CRRTriggerRestore(loader=self)
```
Comment (evelyn-ys) [inline src/dataprotection/azext_dataprotection/manual/commands.py:83]: Can you clear these codes which are commented out?

**Comment by evelyn-ys** [comment]
There are some breaking changes which are not documented
image
Can you confirm it's necessary to make these breaking changes?

**Comment by zubairabid** [comment]
> There are some breaking changes which are not documented image Can you confirm it's necessary to make these breaking changes?

Removal of `dataprotection backup-instance update` is intentional. It had been incorrectly exposed due to misconfiguration in AAZ parameters. I am planning to re-add it with a better options in a PR later this month.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**